src/client/app/effects/snippet.ts

```diff
@@ -288,6 +288,13 @@ export class SnippetEffects {
             case Snippet.ImportType.SAMPLE:
             case Snippet.ImportType.URL:
             case Snippet.ImportType.GIST:
+                let viewUrlMatch = new RegExp(`^${environment.current.config.editorUrl}\/#\/view`);
+                if (type === Snippet.ImportType.URL && viewUrlMatch.test(data)) {
+                    /* If importing from a view-mode URL, simply redirect the user */
```

Comment:
Right now the view mode has access to Angular that parses the URL and gives it access to the param, such as host, gist vs sample, id, etc.. If we wanted to do it in the actual import feature, we would need our own regex matching and URL validation, in addition to the view mode code that already does this. Not sure if we want to do that?